src/metrics/mod.rs

```diff
@@ -16,68 +19,197 @@ pub static PERCENTILES: &[(&str, f64)] = &[
     ("p9999", 99.99),
 ];
 
-pub struct Snapshot {
-    pub prev: HashMap<Metrics, u64>,
+pub struct HistogramsSnapshot {
+    pub previous: HashMap<String, metriken::histogram::Snapshot>,
+    pub deltas: HashMap<String, metriken::histogram::Snapshot>,
+}
+
+pub struct MetricsSnapshot {
```

Comment:
Just to make sure I understand: when the admin thread serves out snapshotted metrics from METRICS_SNAPSHOT, both histogram and counter values will be from the snapshot (and potentially upto 1s stale), as opposed to the previous version where counters would be fresh and the histogram from the snapshot, right?

Reply:
No. The admin thread will still serve fresh counters + gauges, but stale percentiles (with timestamps for percentile metrics only). See L#272 in admin/mod.rs.

This means that each polling agent may see the counters at different points in time. I don't have a strong objection to serving counters + gauges from the metrics snapshot, but would prefer to increase the refresh frequency.

Reply:
Ack. I'm fine with it the way it is -- was mostly just trying to wrap my head around the semantics of what to expect.